packages/server/src/object/index.rs

```diff
@@ -260,6 +262,32 @@ impl Server {
 			}
 		}
 
+		if depth.is_none() {
+			// Attempt to compute the depth.
+			let depth = children_metadata
+				.iter()
+				.filter_map(|option| option.as_ref().and_then(|metadata| metadata.depth))
+				.max();
+
+			// Set the depth if possible.
+			if depth.is_some() {
+				// Set the depth.
+				let p = connection.p();
+				let statement = formatdoc!(
+					"
+						update objects
+						set depth = {p}1
+						where id = {p}2;
+					"
+				);
+				let params = db::params![depth.unwrap() + 1, id];
```

Comment:
Won't you hit this unwrap for objects with no children? I think you want `depth.map_or(1, |depth| depth + 1)`.